### PR TITLE
Animations

### LINKED ISSUE #8: Animation configuration
Allow developer to choose if animations should be applied, if yes, then allow him to provide custom interpolator and animation duration

### PR DESCRIPTION
**Pull request by matejsemancik**
resolves #8 

### library module
Allow to configure:
- enable / disable animation flag
- animation duration
- animation interpolator

Moved defaults to companion object.

### sample module
Added animation controls:
- enable switch
- duration seeker
- interpolator chooser

### CODE REVIEW

File: sample/src/main/kotlin/com/thefuntasty/donutsample/ui/playground/PlaygroundActivity.kt [hunk 1]
```diff
@@ -2,6 +2,7 @@ package com.thefuntasty.donutsample.ui.playground
 
 import android.os.Bundle
```
Comment (thefuntasty-ops) [inline sample/src/main/kotlin/com/thefuntasty/donutsample/ui/playground/PlaygroundActivity.kt]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/thefuntasty/donut/blob/3b46217b153502a1336d7bd3185bf6de4aad857e/sample/src/main/kotlin/com/thefuntasty/donutsample/ui/playground/PlaygroundActivity.kt#L3"></span>Imports must be ordered in lexicographic order without any empty lines in-between</td>
    </tr>
  </tbody>
</table>

Comment (matejsemancik) [reply sample/src/main/kotlin/com/thefuntasty/donutsample/ui/playground/PlaygroundActivity.kt]: AS puts ` kotlinx.*` imports before `kotlin.*` imports for unknown reason, it's not configurable in kotlin code style

File: library/src/main/kotlin/com/thefuntasty/donut/DonutProgressView.kt [hunk 2]
```diff
@@ -118,6 +127,22 @@ class DonutProgressView @JvmOverloads constructor(
             invalidate()
         }
 
+    /**
+     * If true, view will animate changes when new data is submitted.
+     * If false, state change will happen instantly.
+     */
+    var animationEnabled: Boolean = DEFAULT_ANIM_ENABLED
```
Comment (skywall) [inline library/src/main/kotlin/com/thefuntasty/donut/DonutProgressView.kt]: `animationEnable` -> `animateChanges` ?

Comment (matejsemancik) [reply library/src/main/kotlin/com/thefuntasty/donut/DonutProgressView.kt]: makes sense, will change in next PR

**Pull request merged**